app/src/main/java/org/covidwatch/android/data/backend/firebase/FirebaseContactEventFetcher.kt

```diff
@@ -0,0 +1,104 @@
+package org.covidwatch.android.data.backend.firebase
+
+import android.content.Context
+import android.util.Log
+import androidx.work.ListenableWorker
+import com.google.android.gms.tasks.Continuation
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.DocumentChange
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.QuerySnapshot
+import kotlin.ranges.ClosedRange
+import org.covidwatch.android.R
+import org.covidwatch.android.data.CovidWatchDatabase
+import org.covidwatch.android.data.backend.ContactEventFetcher
+import org.covidwatch.android.data.backend.BackendConstants
+import org.covidwatch.android.data.backend.InfectionState
+import java.util.*
+
+/**
+ * Firebase implementation of the ContactEventFetcher interface.
+ */
+class FirebaseContactEventFetcher(
+    private val context: Context
+) : ContactEventFetcher {
+
+    companion object {
+        private const val TAG = "ResultFetcher"
+    }
+
+    override fun fetch(
+        timeWindow: ClosedRange<Date>,
+        markLocalContactEventsCb: List<String>.(InfectionState) -> Unit
+    ) {
+        val task = FirebaseFirestore.getInstance()
+            .collection(BackendConstants.COLLECTION_CONTACT_EVENTS)
+            .whereGreaterThan(
+                BackendConstants.FIELD_TIMESTAMP,
+                Timestamp(timeWindow.start)
+            )
+            .get()
+            .continueWith(
+                CovidWatchDatabase.databaseWriteExecutor,
+                Continuation<QuerySnapshot, Unit> {
+                    it.result?.handleQueryResult(timeWindow, markLocalContactEventsCb)
+                })
+
+
+        // TODO #14 RE-ADD THIS AFTER FIREBASE QUOTA INCREASED
```

Comment:
Delete this comment of mine, no longer meaningful.

Reply:
Done.